index.js

```diff
@@ -26,14 +26,26 @@ function defaultConfig() {
 
 /**
  * Ember Addon, generate resized images on build time
- *
- *
  */
 module.exports = {
   name: 'ember-responsive-image',
   options: {},
   metaData: {},
   app: null,
+  metaExtensions: [],
+
+  /**
+   * This hook will be called after the metaData generation in the postprocessTree-hook.
+   * If you need to change or extend the metaData, you can implement this method in your addon.
+   *
+   * @method addMetaData
+   * @param {Object} metaData
+   * @return {Object} the changed metaData
+   * @public
+   */
+  addMetaData(metaData) {
+    return metaData;
```

Comment:
Your intention is that another addon overrides (patches) this method? Seems a bit weird, that another object is patching this one. Also does not allow for multiple addons to hook into the process...

Maybe something like `addMetaDataHook(fn)`, that adds functions to an internal array, and processes them iteratively:
```js
processMetaData(metaData) {
  this._hooks.forEach(fn => metaData = fn(metaData));
  return metaData;
}
```

Reply:
Sure, you're right. That was also my first approach... see 2 commit before...
For some reason i thought, it looks cleaner this way...

Reply:
Ah, yes, nice. I like that commit better ;)

`target` support is nice, but not really required I think, as nowadays you can easily "fix" the `this` scope, using arrow functions or `Function#bind()`. The implementation can be simplified without it. But it's fine either way!